File: data/derived_cot/rq3_prompting/results/translation/qwen/py_to_cpp/code_output/LongestWord.cpp
#include <cctype>
#include <locale>
#include <string>
#include <vector>
#include <sstream>
#include <algorithm>

class LongestWord {
private:
    std::vector<std::string> word_list;

public:
    void add_word(const std::string& word) {
        word_list.push_back(word);
    }

    std::string find_longest_word(const std::string& sentence) {
        std::string longest_word = "";

        // Convert the entire sentence to lowercase
        std::string sentence_lower;
        std::locale loc;
        for (char c : sentence) {
            sentence_lower += std::tolower(c, loc);
        }

        // Define punctuation characters to remove
        std::string punctuation = "!\"#$%&'()*+,-./:;<=>?@[\\]^_`{|}~";
        std::string sentence_no_punct;
        for (char c : sentence_lower) {
            if (punctuation.find(c) == std::string::npos) {
                sentence_no_punct += c;
            }
        }

        // Split the sentence by spaces
        std::istringstream iss(sentence_no_punct);
        std::string word;
        while (iss >> word) {
            // Check if the word exists in the word_list
            if (std::find(word_list.begin(), word_list.end(), word) != word_list.end()) {
                if (word.length() > longest_word.length()) {
                    longest_word = word;
                }
            }
        }

        return longest_word;
    }
};

File: data/derived_cot/rq1_traces/translation/glm5.1/cpp_to_py/code_output/LongestWord.py
import string

class LongestWord:
    def __init__(self):
        self.word_list = []

    def add_word(self, word: str) -> None:
        self.word_list.append(word)

    def find_longest_word(self, sentence: str) -> str:
        # Convert to lowercase
        lower_sentence = sentence.lower()
        
        # Remove punctuation (equivalent to std::ispunct in C locale)
        translator = str.maketrans('', '', string.punctuation)
        lower_sentence = lower_sentence.translate(translator)
        
        # Split by whitespace (equivalent to std::istringstream >> word)
        words = lower_sentence.split()
        
        # Find the longest word that exists in the word_list
        longest_word = ""
        for word in words:
            # `word in self.word_list` behaves like std::find != end
            if word in self.word_list and len(word) > len(longest_word):
                longest_word = word
                
        return longest_word

    def get_word_list(self) -> list:
        # Return a copy to mimic C++ returning by value (prevents external mutation)
        return list(self.word_list)